tests/feed/test_feed.py

```diff
@@ -0,0 +1,176 @@
+    post = Post(course_id=persist_course,
+                user_id=persist_user,
+                parent_post_id=None,
+                content="this is a root post by a student for first course")
+    post.save()
+    return post
+
+
+@pytest.fixture
+def persist_reply_for_first_course(persist_course, persist_teacher,
+                                   persist_post_for_first_course, persist_second_student_course):
+    post = Post(course_id=persist_course,
+                user_id=persist_teacher,
+                parent_post_id=persist_post_for_first_course,
+                content="this is a reply post by a teacher for message in first course")
+    post.save()
+    return post
+
+
+@pytest.fixture
+def persist_post_for_second_course_student(persist_second_course, persist_second_user):
+    post = Post(course_id=persist_second_course,
+                user_id=persist_second_user,
+                parent_post_id=None,
+                content="this is a root post by a second student for second course")
+    post.save()
+    return post
+
+
+@pytest.mark.django_db
+class TestPost:
+    def test_is_root_post(self, persist_post_for_first_course, persist_reply_for_first_course):
+        assert persist_post_for_first_course.is_root_post()
+        assert not persist_reply_for_first_course.is_root_post()
+
+    def test_post_is_only_reply_to_root_post(self, persist_user, persist_course,
+                                             persist_reply_for_first_course):
+        reply_to_reply = Post(course_id=persist_course, user_id=persist_user,
+                              parent_post_id=persist_reply_for_first_course, content="should be invalid")
+        with pytest.raises(ValidationError):
+            reply_to_reply.save()
+
+    def test_is_user_able_to_post_in_course(self, persist_user, persist_course):
+        assert not Post.is_user_able_to_post_in_course(persist_user, persist_course)
+        user_joined_group = StudentCourses(student_id=persist_user, teacher_course_id=persist_course,
+                                           status="Confirmed")
+        user_joined_group.save()
+        assert Post.is_user_able_to_post_in_course(persist_user, persist_course)
+
+    def create_post_for_user_not_related_to_course(self, persist_user, persist_course):
+        assert not Post.is_user_able_to_post_in_course(persist_user, persist_course)
+        invalid_post = Post(course_id=persist_course, user_id=persist_user, content="should be invalid")
+        with pytest.raises(ValidationError):
+            invalid_post.save()
+
+    def test_image_upload(self, image_file, persist_post_for_first_course):
+        persist_post_for_first_course.image = image_file
+        persist_post_for_first_course.save()
+        media_path = os.path.join(settings.MEDIA_ROOT, str(persist_post_for_first_course.image))
+
+        assert persist_post_for_first_course.image.read() == b"file_content"
+        assert os.path.isfile(media_path)
```

Comment:
Why do you need this check? if it's saved in the DB, won't it be saved in `MEDIA_ROOT` for sure?

Reply:
If i understand correctly images are not saved to DB using imageField, but to file system only,
I wanted to make sure I configured everything correctly and the file exists in media root after save.
 I guess the first check to read bytes is redundant

Reply:
Usually I recommend using URLs to images in those kind of projects as storing real images is complicated, but seems like you made it. good job

Reply:
That being said, if you figured out how is it done, I'm not sure a test is needed, unless you check for specifics regarding the image saving functionality. then I'd think of a way to maybe separate this test from the Feed app, as it may be used in different parts of the whole application.

Reply:
It does make more sense this test being outside the feed app tests, I'll move it to a different .py file inside the tests directory

Reply:
Also, can it be part of this PR?

Reply:
If it's relevant for this PR - yes.
If that makes the PR too big, maybe it should be in a different PR